lib/winapi/filemanip.c

```diff
@@ -241,6 +246,145 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
     }
 }
 
+BOOL CopyFileA (LPCSTR lpExistingFileName, LPCSTR lpNewFileName, BOOL bFailIfExists)
+{
+    NTSTATUS status;
+    HANDLE sourceHandle;
+    HANDLE targetHandle = INVALID_HANDLE_VALUE;
+    ANSI_STRING sourcePath;
+    ANSI_STRING targetPath;
+    OBJECT_ATTRIBUTES objectAttributes;
+    IO_STATUS_BLOCK ioStatusBlock;
+    FILE_BASIC_INFORMATION fileBasicInformation;
+    FILE_NETWORK_OPEN_INFORMATION networkOpenInformation;
+    LPVOID readBuffer = NULL;
+    SIZE_T readBufferRegionSize = 64 * 1024;
+    DWORD bytesRead;
+
+    sourceHandle = CreateFile(
+        lpExistingFileName,
+        FILE_GENERIC_READ,
+        0,
```

Comment:
You can use `FILE_SHARE_READ` here instead, that will allow other threads to read from the same file, while still preventing it from getting modified.

Reply:
Done.